azure-pipelines.yaml

```diff
@@ -7,103 +7,88 @@ pool:
-    go env
-  displayName: 'Set up the Go workspace'
-  env:
-    GO111MODULE: on
+  - script: |
+      docker info
+      docker load -i ./dockercache/devcontainer.tar
+    condition: eq(variables.DOCKER_CACHE_HIT, 'true')
+    displayName: Load cached devcontainer image
 
-- script: |
-    set -x
-    mkdir -p '$(MODULE_PATH)/bin'
-    os=$(go env GOOS)
-    arch=$(go env GOARCH)
-    # download kubebuilder
-    curl -sL https://go.kubebuilder.io/dl/2.2.0/${os}/${arch} | tar -xz -C $(MODULE_PATH)/bin
-    mv $(MODULE_PATH)/bin/kubebuilder_2.2.0_${os}_${arch}/bin/* $(MODULE_PATH)/bin/
-    # download kustomize
-    curl -sL https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv3.4.0/kustomize_v3.4.0_${os}_${arch}.tar.gz | tar -xz -C /tmp/
-    mv /tmp/kustomize $(MODULE_PATH)/bin/kustomize
-    # set permission
-    chmod a+x $(MODULE_PATH)/bin/*
-    # export path
-    export PATH=$PATH:$(MODULE_PATH)/bin
-    # Install golangci-linter
-    curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.21.0
-    # install test dependencies
-    go mod download
-    make install-test-dependency
-    # run test
-    make test
-  continueOnError: 'false'
-  displayName: 'Install kubebuilder, kustomize and Run unit tests'
-  env:
-    GO111MODULE: on
-    KUBEBUILDER_ASSETS: $(MODULE_PATH)/bin
-    DATABRICKS_HOST: $(DATABRICKS_HOST)
-    DATABRICKS_TOKEN: $(DATABRICKS_TOKEN)
-  workingDirectory: '$(MODULE_PATH)'
+  - script: |
+      # Copy the makefile into the container folder
+      cp ./Makefile ./.devcontainer/Makefile
 
-  
-- task: PublishCodeCoverageResults@1
-  inputs:
-    codeCoverageTool: Cobertura 
-    summaryFileLocation: $(System.DefaultWorkingDirectory)/**/cover.xml
-    reportDirectory: $(System.DefaultWorkingDirectory)/**/cover
+      echo "-------> Building devcontainer"
+      time docker build --cache-from devcontainer:latest -t devcontainer -f ./.devcontainer/Dockerfile ./.devcontainer
+      # Create a directory for go mod cache
+      mkdir -p $(System.DefaultWorkingDirectory)/.gocache
 
+      echo "-------> Building code and running tests"
+      # Run `make` to build and test the code
+      time docker run -v $(System.DefaultWorkingDirectory)/.gocache:/go/pkg/ -v /var/run/docker.sock:/var/run/docker.sock -v $(System.DefaultWorkingDirectory):/src --workdir /src --entrypoint /bin/bash --network="host" --env DATABRICKS_HOST=$(DATABRICKS_HOST) --env DATABRICKS_TOKEN=$(DATABRICKS_TOKEN) devcontainer -c "make test" 
+    displayName: Build and Test
 
-- script: docker build -t $(IMAGE_NAME) .  # add options to this command to meet your needs
-  condition: and(succeeded(), ne(variables['Build.SourceBranch'], 'refs/heads/master'))
-  displayName: 'Docker build'
-  workingDirectory: '$(MODULE_PATH)'
+  - script: |
+      echo "-------> Saving docker image"
+      time docker image save -o ./.dockercache/devcontainer.tar devcontainer
```

Comment:
seems we need to add  `mkdir -p $(System.DefaultWorkingDirectory)/.dockercache `  
https://dev.azure.com/ms/azure-databricks-operator/_build/results?buildId=63330&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774&t=cb33ee24-f492-54d2-727a-ae0a7bb84b2c&l=23

Reply:
Good catch, strange how that never showed itself last time I ran the pipeline 🤔 will fix up today

Reply:
This should be fixed now 👍